server/src/test/java/org/elasticsearch/cluster/routing/IndexRoutingTests.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch.cluster.routing;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import static org.hamcrest.Matchers.hasSize;
+
+public class IndexRoutingTests extends ESTestCase{
+    public void testCollectSearchShardsInStandardIndex() {
+        for (int shards = 1; shards < 5; shards++) {
+            IndexRouting indexRouting = IndexRouting.build(shards, 1, 1);
+
+            for (int i = 0; i < 20; i++) {
+                String routing = randomUnicodeOfLengthBetween(1, 50);
+
+                Set<Integer> searchShardSet = new HashSet<>();
+                indexRouting.collectSearchShards(routing, searchShardSet::add);
+                assertThat(searchShardSet, hasSize(1));
+            }
+        }
+    }
+
+    public void testPartitionedIndex() {
```

Comment:
This and all tests below it duplication `OperationRoutingTests`. They are *slightly* lower level in that they don't test the creation of the `IndexRouting` from information in the `IndexMetadata`. But they are close. Paranoia drove me to duplicate them rather than move them.

Reply:
Now that I've reworked how this is integrated I've moved these test from `OperationRoutingTests` .